services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/dao/VlanRangeRepository.java

```diff
@@ -0,0 +1,186 @@
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.db.CacheFactory;
+import com.futurewei.alcor.common.db.ICache;
+import com.futurewei.alcor.common.db.Transaction;
+import com.futurewei.alcor.common.exception.*;
+import com.futurewei.alcor.common.repo.ICacheRepository;
+import com.futurewei.alcor.web.allocator.VlanKeyAllocator;
+import com.futurewei.alcor.web.entity.NetworkVlanRange;
+import com.futurewei.alcor.web.entity.NetworkVlanRangeRequest;
+import com.futurewei.alcor.web.entity.NetworkVlanType;
+import com.futurewei.alcor.web.entity.VlanKeyAlloc;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+@ComponentScan(value="com.futurewei.alcor.common.db")
+public class VlanRangeRepository implements ICacheRepository<NetworkVlanRange> {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public ICache<String, NetworkVlanRange> getCache() {
+        return cache;
+    }
+
+    private ICache<String, NetworkVlanRange> cache;
+
+
+    @Autowired
+    public VlanRangeRepository(CacheFactory cacheFactory) {
+        cache = cacheFactory.getCache(NetworkVlanRange.class);
+    }
+
+    @PostConstruct
+    private void init() {
+        logger.info( "VlanRangeRepository init completed");
+    }
+
+    @Override
+    public synchronized NetworkVlanRange findItem(String id) {
+        try {
+            return cache.get(id);
+        } catch (CacheException e) {
+            e.printStackTrace();
+            logger.error("VlanRangeRepository findItem() exception:", e);
+        }
+        return null;
+    }
+
+    @Override
+    public synchronized Map<String, NetworkVlanRange> findAllItems() {
+        try {
+            return cache.getAll();
+        } catch (CacheException e) {
+            e.printStackTrace();
+            logger.error("VlanRangeRepository findAllItems() exception:", e);
+        }
+
+        return new HashMap();
+    }
+
+    @Override
+    public synchronized void addItem(NetworkVlanRange networkVlanRange){
+        logger.error("Add networkVlanRange:{}", networkVlanRange);
+
+        try {
+            cache.put(networkVlanRange.getId(), networkVlanRange);
+        } catch (CacheException e) {
+            e.printStackTrace();
+            logger.error("VlanRangeRepository addItem() exception:", e);
+        }
+    }
+
+    @Override
+    public synchronized void deleteItem(String id) {
+        logger.error("Delete rangeId:{}", id);
+
+        try {
+            cache.remove(id);
+        } catch (CacheException e) {
+            e.printStackTrace();
+            logger.error("VlanRangeRepository deleteItem() exception:", e);
+        }
+    }
+
+    /**
```

Comment:
Like the comment! Can we add it everywhere :-)

Reply:
Sure!